include/deal.II/lac/precondition.h

```diff
@@ -767,61 +816,171 @@ namespace internal
     const bool has_Tstep<T, VectorType>::value;
 
     template <
+      typename MatrixType,
       typename PreconditionerType,
       typename VectorType,
       typename std::enable_if<has_step<PreconditionerType, VectorType>::value,
                               PreconditionerType>::type * = nullptr>
     void
-    step(const PreconditionerType &preconditioner,
+    step(const MatrixType &,
+         const PreconditionerType &preconditioner,
          VectorType &              dst,
-         const VectorType &        src)
+         const VectorType &        src,
+         const double              relaxation,
+         VectorType &,
+         VectorType &)
     {
+      Assert(relaxation == 1.0, ExcInternalError());
+
+      (void)relaxation;
```

Comment:
This feels a bit unfortunate that we can't embed the relaxation factor here. I did not look into all details, but I wonder whether this is a technical limitation (i.e., difficult to embed into typical preconditioners) or an interface limitation (i.e., backward compatibility to existing preconditioners).

Reply:
This is not a limitation. The relaxation is for the 4 existing "preconditioners" in the responsibility of the preconditioner/system matrix (see also #13152). This was the only way to get everything under one hood.